File: typescript/src/common/application/policy.ts
import type { Command } from './command.js';
import type { AggregateId } from '../domain/identity/aggregateId.js';
import type { Event } from '../domain/model/event.js';

export interface Policy<E extends Event<string, AggregateId>, C extends Command<string, AggregateId>> {
  evaluate(event: E): C | null;
}

File: java/src/main/java/it/giannibombelli/wsc2026/common/application/Policy.java
package it.giannibombelli.wsc2026.common.application;

import it.giannibombelli.wsc2026.common.application.Command;
import it.giannibombelli.wsc2026.common.domain.model.Event;

public interface Policy<E extends Event<?>, C extends Command<?>> {
    C evaluate(E event);
}
